### DISCUSSION TITLE
Issues when running jest tests

### DISCUSSION THREAD

**Discussion by niyowski** [software-mansion/react-native-reanimated#4319] [Q&A]
I'm not sure whether this is a bug or my configuration is wrong, but I get the errors that I described [here](https://github.com/callstack/react-native-testing-library/discussions/1379).

Error 1:

> <path-to-root>/node_modules/react-native-reanimated/src/reanimated2/mock.ts:8
>     const IMMEDIATE_CB_INVOCATION = (cb: () => unknown) => cb();
>                                        ^
>     SyntaxError: Unexpected token ':'
> 
>       3 |
>       4 | jest.mock('react-native-reanimated', () => {
>       5 |   const Reanimated = require('react-native-reanimated/mock.js'); <--

Error 2:

> node_modules/react-native-reanimated/src/reanimated2/mock.ts:4:1 - error TS1208: 'mock.ts' cannot be compiled under '--isolatedModules' because it is considered a global script file. Add an import, export, or an empty 'export {}' statement to make it a module.
>
>    4 const NOOP = () => {
>      ~~~~~

Error 3:

>   <path-to-root>/node_modules/@react-native/polyfills/error-guard.js:14
>    type ErrorHandler = (error: mixed, isFatal: boolean) => void;
>         ^^^^^^^^^^^^
>
>    SyntaxError: Unexpected identifier

jest.config.json:

```js
const { defaults: tsjPreset } = require('ts-jest/presets');

/** @type {import('ts-jest').JestConfigWithTsJest} */
module.exports = {
  ...tsjPreset,
  preset: 'react-native',
  transform: {
    '^.+\\.jsx$': 'babel-jest',
    '^.+\\.tsx?$': [
      'ts-jest',
      {
        tsconfig: 'tsconfig.spec.json',
      },
    ],
  },
  // Produces Error 1 when transformIgnorePatterns is not configured
  transformIgnorePatterns: [
    // Produces Error 2
    'node_modules/(?!(jest-)?@?react-native|@react-native-community|@react-navigation)',
    // Produces Error 3
    'node_modules/(?!(jest-)?react-native|@react-native-community|@react-navigation)',
  ],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  moduleNameMapper: {
    '^~/(.*)$': '<rootDir>/src/$1',
    '^test/(.*)$': '<rootDir>/test/$1',
    '\\.(png|jpg|ico|jpeg|gif|eot|otf|webp|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mock__/AssetMock.js',
    '\\.svg': '<rootDir>/__mocks__/svgMock.js',
  },
  setupFilesAfterEnv: ['<rootDir>/test/setup.ts'],
};
```

test/setup.ts:

```ts
import '@testing-library/jest-native/extend-expect';
import 'react-native-gesture-handler/jestSetup';

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock.js');
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
```

tsconfig.spec.json:

```json
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "jsx": "react"
  }
}
```

How could I possibly solve this? I'll appreciate your suggestions.

**Comment by tjzel** [comment]
Hi, could you check with `'node_modules/(?!((jest-)?react-native(-.*)?|@react-native(-community)?|@react-navigation|@rneui)/)'`?

**Reply by tjzel** [reply]
@niyowski just bumping if you didn't get notification about this

**Reply by myou11** [reply]
Hi @tjzel, I tried this and it did not work. Still getting the `'mock.ts' cannot be compiled` error using your string in `transformIgnorePatterns`
